src/pages/MainPage.tsx

```diff
@@ -60,8 +60,7 @@ const MainPage: React.FC = () => {
 
   const KakaoLoginBtn = () => {
     const location = useLocation();
-    const from = location.state?.from?.pathname || "/group";
-    const redirectUrl = `${baseUrl}${from}`;
+    const groupId = location.state?.from?.pathname.split("/")[2] || "";
```

Comment:
@myeongjunkim  
(수정사항)split으로 groupId를 받고 없으면 빈값으로 보내서 TermServicePage에서 null로 인지하게끔 합니다!

Reply:
```suggestion
  const pathname = location.state?.from?.pathname || '';

// 슬래시(/)를 기준으로 나누기
const pathParts = pathname.split('/');

// 경로가 '/group/{groupId}' 형식인지 확인
const groupId = pathParts.length === 3 && pathParts[1] === 'group' ? pathParts[2] : null;

console.log(groupId); // groupId가 존재하면 출력, 없으면 null
```


이런식으로 해야될 것 같아요! 나중에 /group/groupId 이외의 페이지에서 리다이렉트 시킬 경우 인덱스가 터질 수도 있을 것 같습니다.